src/components/DynamicWidgets.js

```diff
@@ -0,0 +1,87 @@
+import { createWidgetMixin } from '../mixins/widget';
+import { EXPERIMENTAL_connectDynamicWidgets } from 'instantsearch.js/es/connectors';
+import { createSuitMixin } from '../mixins/suit';
+
+function getWidgetAttribute(vnode) {
+  const props = vnode.componentOptions && vnode.componentOptions.propsData;
+  if (props) {
+    if (props.attribute) {
+      return props.attribute;
+    }
+    if (Array.isArray(props.attributes)) {
+      return props.attributes[0];
+    }
+  }
+
+  const children =
+    vnode.componentOptions && vnode.componentOptions.children
+      ? vnode.componentOptions.children
+      : vnode.children;
+
+  if (Array.isArray(children)) {
+    // return first child with a truthy attribute
+    return children.reduce(
+      (acc, curr) => acc || getWidgetAttribute(curr),
+      undefined
+    );
```

Comment:
```suggestion
    for (let child of children) {
      const attribute = getWidgetAttribute(child);
      if (attribute) {
        return attribute;
      }
    }
```

kinda old school but easier to read along with that comment, IMO.

Reply:
for of needs transpilation, what's wrong with the reduce?

Reply:
didn't think of transpilation. nothing wrong with the reduce, just it took me several seconds to follow where it's going.